applications/ehr_query_llm/lmm/riva_asr.py

```diff
@@ -0,0 +1,274 @@
+        self.done_speaking = False
+        # Queue used to write ASR responses to in a background thread
+        self.streaming_queue = Queue()
+        # Used to mute the mic
+        self.is_muted = True
+        # Flag that gives Riva time to output final prediction
+        self.asr_service = riva.client.ASRService(auth)
+        self.auth = auth
+        self.stub = rasr_srv.RivaSpeechRecognitionStub(auth.channel)
+        self.input_queue = AudioQueue()
+
+        streaming_config = riva.client.StreamingRecognitionConfig(
+            config=riva.client.RecognitionConfig(
+                encoding=riva.client.AudioEncoding.LINEAR_PCM,
+                language_code=cli_args.language_code,
+                max_alternatives=1,
+                profanity_filter=cli_args.profanity_filter,
+                enable_automatic_punctuation=True,
+                verbatim_transcripts=not cli_args.no_verbatim_transcripts,
+                sample_rate_hertz=cli_args.sample_rate_hz,
+                audio_channel_count=1,
+            ),
+            interim_results=True,
+        )
+        self.streaming_config = streaming_config
+        riva.client.add_word_boosting_to_config(
+            streaming_config, boosted_lm_words, boosted_lm_score
+        )
+        self.sample_rate_hz = cli_args.sample_rate_hz
+        self.whisper_response = ""
+        # Used to store input microphone audio bytes
+        self.audio_bytes = bytes()
+        self.whisper_pipeline = self._get_whisper_pipeline()
+        super().__init__(fragment, *args, **kwargs)
+
+    @staticmethod
+    def _get_whisper_pipeline():
+        """
+        Function that returns a pipeline for Whisper ASR
+        """
+
+        device = "cuda:0"
+        torch_dtype = torch.float16
+
+        model_id = "openai/whisper-medium.en"
+
+        model = AutoModelForSpeechSeq2Seq.from_pretrained(
+            model_id,
+            torch_dtype=torch_dtype,
+            low_cpu_mem_usage=True,
+            use_safetensors=True,
+        )
+        model.to(device)
+
+        processor = AutoProcessor.from_pretrained(model_id)
+        pipe = pipeline(
+            "automatic-speech-recognition",
+            model=model,
+            tokenizer=processor.tokenizer,
+            feature_extractor=processor.feature_extractor,
+            max_new_tokens=128,
+            torch_dtype=torch_dtype,
+            device=device,
+        )
+        return pipe
+
+    def _get_whisper_transcription(self):
+        """
+        Function that returns a transcription from Whisper ASR
+        """
+        audio_data = np.frombuffer(self.audio_bytes, dtype=np.int16).flatten().astype(np.float32)
+        audio_data /= np.iinfo(np.int16).max
+        audio_data = self._resample_to_16k(audio_data, self.sample_rate_hz)
+        transcription = self.whisper_pipeline(audio_data)["text"]
+        # Clear the
+        self.audio_bytes = bytes()
```

Comment:
Comment needs update?

Reply:
Removed the half deleted comment which was not needed really as the code itself is self-explanatory. Corrected in the next commit.